docs/integrations/authentication/auth0/auth0-demo-app.mdx

```diff
@@ -0,0 +1,254 @@
+
+**Note:** While this article does not require you to have [Next.JS knowledge](https://nextjs.org/), it requires an Auth0 account with a basic configuration, and Node.JS installed in your [local](https://nodejs.org/en/download) or [remote](https://codesandbox.io/signin?continue=/dashboard?import_repo=true) environment.
+
+You can see more information about [RBAC Permit and Auth0](https://www.permit.io/blog/add-rbac-authorization-to-auth0) in this great blog post by Gabriel L. Manor
+
+The full code of the demo application is available [here](https://github.com/permitio/permit-next-todo-starter/tree/auth0-integration)
+
+For our example App, we will create the following roles (it is recommended to use lowercase keys, as keys are case-sensitive):
+
+  * admin
+  * manager
+  * viewer
+ 
+### 1.2. Create the right resources in Permit
+In our example, we will set up a todo application, so we will create a `task` resource with the following actions:
+
+- get
+- post
+- delete
+
+Setting up roles and resources can be done through the [Permit dashboard](https://app.permit.io/policy-editor/roles), by using the [Permit API](https://api.permit.io/v2/redoc#tag/Roles/operation/create_role), or with one of our SDKs: [Python](https://docs.permit.io/sdk/python/usage-example/#create-a-role), [NodeJS](https://docs.permit.io/sdk/nodejs/usage-example#create-a-role) and more. Additional SDKs are available in the SDK Reference section [inside the docs](https://docs.permit.io/).
+
+### 1.3. Set up the right permissions in Permit.io:
+
+- `admin` should have all the permissions for the `task` resource.
+- `manager` will only have `get` and `post` permissions.
+- `viewer` will only have the `get` permission.
+
+As noted earlier, these permissions can be set up through either the [Permit dashboard](https://app.permit.io/policy-editor) or by using the [Permit API](https://api.permit.io/v2/redoc#tag/Roles/operation/assign_permissions_to_role).
+
+## 2. Run the demo application
+### 2.1. Clone the demo application:
+
+```bash
+git clone https://github.com/permitio/permit-next-todo-starter
+git checkout auth0-integration
+```
+
+### 2.2. Put your Auth0 credentials in the .env file
+To get your own keys in the desired format, [use this link](https://auth0.com/docs/quickstart/webapp/nextjs/interactive#configure-the-sdk) while logged in to your Auth0 account
+
+```bash
+AUTH0_SECRET='<auth0_secret>'
+AUTH0_BASE_URL='http://localhost:3000'
+AUTH0_ISSUER_BASE_URL='<auth0_issuer_base_url>'
+AUTH0_CLIENT_ID=<auth0_client_id>
+AUTH0_CLIENT_SECRET='<auth0_client_secret>'
+```
+
+### 2.3. Put your Permit credentials in the .env file
+You can get your Permit API Key from the [projects page](https://app.permit.io/projects) by clicking on the three dots on the top right of your environment and selecting `Copy API Key`.
+
+_![Get Permit API key](/img/onboarding/secret-key-1.png)_
+
+```bash
+PERMIT_API_KEY=<permit_api_key>
+```
+
+
+
+## 3. Demo application explanation
```

Comment:
Please find a different title here - this cannot be the same as the first section and the entire page.